### ISSUE TITLE
[BUG] Avoid InlineResponse

### ISSUE THREAD

**Issue by sbilello** [OpenAPITools/openapi-generator#9485]
#### Bug Report Checklist

- [X] Have you provided a full/minimal spec to reproduce the issue?
- [X] Have you validated the input using an OpenAPI validator ([example](https://apidevtools.org/swagger-parser/online/))?
- [ ] Have you [tested with the latest master](https://github.com/OpenAPITools/openapi-generator/wiki/FAQ#how-to-test-with-the-latest-master-of-openapi-generator) to confirm the issue still exists?
- [X] Have you searched for related issues/PRs?
- [ ] What's the actual output vs expected output?
- [ ] [Optional] Sponsorship to speed up the bug fix or feature request ([example](https://github.com/OpenAPITools/openapi-generator/issues/6178))

##### Description

How can I avoid the InlineResponse classes and have proper names?

##### OpenAPI declaration file

```yaml
openapi: 3.0.1
info:
  contact:
    email: sergio.bilello@test.us
  title: meetings
  description: meetings API
  version: 1.0.0
servers:
  - url: 'https://test.us'
components:
  schemas: {}
  securitySchemes:
    bearerAuth:
      name: Authorization
      type: apiKey
      in: header
  responses:
    MeetingList:
      description: Meeting List
      content:
        application/json:
          schema:
            description: ''
            type: object
            properties:
              page_count:
                type: number
              page_number:
                type: number
              page_size:
                type: number
              total_records:
                type: number
              meetings:
                type: array
                uniqueItems: true
                minItems: 1
                items:
                  required:
                    - uuid
                    - id
                    - host_id
                    - topic
                    - type
                    - start_time
                    - duration
                    - timezone
                    - created_at
                    - join_url
                  properties:
                    uuid:
                      type: string
                      minLength: 1
                    id:
                      type: number
                    host_id:
                      type: string
                      minLength: 1
                    topic:
                      type: string
                      minLength: 1
                    type:
                      type: number
                    start_time:
                      type: string
                      minLength: 1
                    duration:
                      type: number
                    timezone:
                      type: string
                      minLength: 1
                    created_at:
                      type: string
                      minLength: 1
                    join_url:
                      type: string
                      minLength: 1
            required:
              - page_count
              - page_number
              - page_size
              - total_records
              - meetings
            x-examples:
              example-1:
                page_count: 50385
                page_number: 1
                page_size: 1
                total_records: 50385
                meetings:
                  - uuid: testUUID
                    id: 892271555
                    host_id: testhostID
                    topic: Meeting For TEST
                    type: 8
                    start_time: '2020-05-14T13:02:04Z'
                    duration: 60
                    timezone: America/Phoenix
                    created_at: '2020-05-14T14:02:01Z'
                    join_url: 'https://test.us/j/2'
    PastMeeting:
      description: Example response
      content:
        application/json:
          schema:
            description: ''
            type: object
            properties:
              uuid:
                type: string
                minLength: 1
              id:
                type: number
              host_id:
                type: string
                minLength: 1
              type:
                type: number
              topic:
                type: string
                minLength: 1
              user_name:
                type: string
                minLength: 1
              user_email:
                type: string
                minLength: 1
              start_time:
                type: string
                minLength: 1
              end_time:
                type: string
                minLength: 1
              duration:
                type: number
              total_minutes:
                type: number
              participants_count:
                type: number
              dept:
                type: string
                minLength: 1
              source:
                type: string
                minLength: 1
            required:
              - uuid
              - id
              - host_id
              - type
              - topic
              - user_name
              - user_email
              - start_time
              - end_time
              - duration
              - total_minutes
              - participants_count
              - dept
              - source
            x-examples:
              example-1:
                uuid: test==
                id: 18971
                host_id: test
                type: 2
                topic: My Meeting
                user_name: API1 TA2
                user_email: ciao@ciao.com
                start_time: '2020-04-01T07:00:30Z'
                end_time: '2020-04-01T07:02:01Z'
                duration: 2
                total_minutes: 2
                participants_count: 1
                dept: web-api
                source: 000M
    MeetingTimeFrame:
      description: Example response
      content:
        application/json:
          schema:
            description: ''
            type: object
            properties:
              from:
                type: string
                minLength: 1
              to:
                type: string
                minLength: 1
              page_count:
                type: number
              page_size:
                type: number
              total_records:
                type: number
              next_page_token:
                type: string
                minLength: 1
              meetings:
                type: array
                uniqueItems: true
                minItems: 1
                items:
                  required:
                    - uuid
                    - id
                    - topic
                    - host
                    - email
                    - user_type
                    - start_time
                    - end_time
                    - duration
                    - participants
                    - has_pstn
                    - has_voip
                    - has_3rd_party_audio
                    - has_video
                    - has_screen_share
                    - has_recording
                    - has_sip
                  properties:
                    uuid:
                      type: string
                      minLength: 1
                    id:
                      type: number
                    topic:
                      type: string
                      minLength: 1
                    host:
                      type: string
                      minLength: 1
                    email:
                      type: string
                      minLength: 1
                    user_type:
                      type: string
                      minLength: 1
                    start_time:
                      type: string
                      minLength: 1
                    end_time:
                      type: string
                      minLength: 1
                    duration:
                      type: string
                      minLength: 1
                    participants:
                      type: number
                    has_pstn:
                      type: boolean
                    has_voip:
                      type: boolean
                    has_3rd_party_audio:
                      type: boolean
                    has_video:
                      type: boolean
                    has_screen_share:
                      type: boolean
                    has_recording:
                      type: boolean
                    has_sip:
                      type: boolean
            required:
              - from
              - to
              - page_count
              - page_size
              - total_records
              - next_page_token
              - meetings
            x-examples:
              example-1:
                from: '2021-04-15'
                to: '2021-05-15'
                page_count: 9
                page_size: 1
                total_records: 9
                next_page_token: qGfwYaCveroiky7vSJs3oYQwLNgU3U3RM32
                meetings:
                  - uuid: yVaEafxxRN+tiOZUSWKJVw==
                    id: 4554919265
                    topic: Test's Personal Meeting Room
                    host:  Test
                    email: usersetting0002@cicicicic.us
                    user_type: Licensed
                    start_time: '2021-04-26T05:24:58Z'
                    end_time: '2021-04-26T05:26:16Z'
                    duration: '01:18'
                    participants: 2
                    has_pstn: false
                    has_voip: false
                    has_3rd_party_audio: false
                    has_video: false
                    has_screen_share: false
                    has_recording: true
                    has_sip: false
security:
  - bearerAuth: []
paths:
  /v2/metrics/meetings:
    get:
      tags:
        - MeetingTimeFrame
      operationId: meetingTimeFrame
      description: Get Meetings in a time frame
      parameters:
        - name: access_token
          in: query
          schema:
            type: string
          example: '{{accessToken}}'
        - name: from
          in: query
          schema:
            type: string
          example: 2020%2D05%2D14
        - name: to
          in: query
          schema:
            type: string
          example: 2021%2D05%2D15
        - name: type
          in: query
          schema:
            type: string
          example: past
        - name: page_size
          in: query
          schema:
            type: string
          example: '1'
      summary: MeetingTimeFrame
      responses:
        '200':
          $ref: '#/components/responses/MeetingTimeFrame'
  '/v2/users/{userId}/meetings':
    get:
      tags:
        - Meetings
      operationId: listMeetings
      description: List of meetings
      parameters:
        - name: userId
          in: path
          required: true
          schema:
            type: string
        - name: type
          in: query
          schema:
            type: string
          example: scheduled
        - name: page_size
          in: query
          schema:
            type: string
          example: '30'
      responses:
        '200':
          $ref: '#/components/responses/MeetingList'
  '/v2/past_meetings/{meetingUUID}':
    get:
      tags:
        - PastMeetings
      operationId: pastMeetings
      description: Auto generated using Swagger Inspector
      parameters:
        - name: meetingUUID
          in: path
          required: true
          schema:
            type: string
      responses:
        '200':
          $ref: '#/components/responses/PastMeeting'
tags:
  - name: MeetingTimeFrame
    description: Get Meetings By TimeFrame
  - name: Meetings
    description: Funny nosed pig-head racoon.
  - name: PastMeetings
    description: Angry short-legged omnivores.

```
##### Generation Details

```
 openapi-generator --version
  openapi-generator-cli 5.1.0
  commit : e023eaa
  built  : 2021-03-20T09:18:49Z
  source : https://github.com/openapitools/openapi-generator
  docs   : https://openapi-generator.tech/
```

##### Steps to reproduce

```
❯ openapi-generator generate -i input.yaml -g java -o output
```

**Comment by wing328** [comment]
Define these schemas separately and use $ref instead, e.g.

https://github.com/OpenAPITools/openapi-generator/blob/master/modules/openapi-generator/src/test/resources/3_0/petstore.yaml#L37

https://github.com/OpenAPITools/openapi-generator/blob/master/modules/openapi-generator/src/test/resources/3_0/petstore.yaml#L45

That way you will complete control of the model naming.

**Comment by sbilello** [comment]
Thanks I changed as you suggested and now I can avoid the InlineResponse prefix :)

**Comment by stonecharioteer** [comment]
Can't we use `#/components/responses/ResponseName` instead? I'm trying to do that, but it still results in InlineBody. To detail, my responses inturn point to the right schema.

**Comment by mustansirali** [comment]
Does this fix apply for version 2.0? I cannot seem to generate the custom object no matter what I do
